source/common/network/connection_impl.cc

```diff
@@ -617,24 +649,33 @@ void ConnectionImpl::setBufferLimits(uint32_t limit) {
   }
 }
 
+void ConnectionImpl::setBufferHighWatermarkTimeout(std::chrono::milliseconds timeout) {
```

Comment:
Since this is a public interface, it is possible that a filter or other places update this, will this work if other places call this method? It looks to me the timer will not be updated or reset in `scheduleBufferHighWatermarkTimeout`.